tools/train.py

```diff
@@ -100,8 +88,8 @@ def parse_args():
         "'--cfg-options model.backbone.depth=18 model.backbone.with_cp=True'")
     parser.add_argument(
         '--launcher',
-        choices=['none', 'pytorch', 'slurm', 'mpi'],
```

Comment:
It seems that this PR introduces a lot of deprecations, which should be avoided. 
1. 'mpi' launch should be kept because we need it when running on parrots in K8s.
2. --gpus and --gpu-ids should be kept for a while with deprecation warning, then we can deprecate them.